wgpu/src/backend/wgpu_core.rs

```diff
@@ -2996,6 +3023,35 @@ fn default_error_handler(err: crate::Error) {
     panic!("wgpu error: {err}\n");
 }
 
+impl From<&CreateShaderModuleError> for CompilationInfo {
```

Comment:
might have missed this earlier, but similar to comment on `impl From<...> for SourceLocation` I'd prefer to avoid `From` implementations that operate on references. From/Into is supposed to take ownership of the thing it converts and callsites can get weird & confusing when it sometimes does and sometimes doesn't. (it still consumes the _reference_, but no longer the object itself)